### ISSUE TITLE
Overwrite `_navigateBack` for Header

### ISSUE THREAD

**Issue by fernandes** [react-community/react-navigation#2372]
### Current Behavior
Trying to overwrite Header._navigateBack so I can use my own function.

```jsx
import { Header } from 'react-navigation'

function overwriteNavigateBack(WrappedComponent) {
  return class extends React.Component {
    _navigateBack() {
      console.log('my navigate');
    };

    render() {
      return <WrappedComponent {...this.props} />;
    }
  }
}

const WrappedHeader = overwriteNavigateBack(Header);

const RouteConfigs = {
  StackOne: {
    screen: StackOne,
    navigationOptions: {
      title: 'Stack One',
      header: (props) => <WrappedHeader {...props} />
    },
  },

StackNavigator(RouteConfigs, StackNavigatorConfig)
```

### Expected Behavior
console.log('my navigate'); should be called, but it's not happening

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation |  1.0.0-beta.12
| react-native     |  0.47.1
| node             | v8.1.2
| npm or yarn      | yarn 0.24.5

ps: I'm new to RN, sorry if missing some information or wrong term used.

//cc @AlanFoster @satya164 I'm mentioning both of you, because you added this function to Header, according to commit https://github.com/react-community/react-navigation/commit/36642ebb69778f5a11f49302987b2a79b470876f so I think you could give me any light here. Any suggestion is really appreciated. Thank you!

**Comment by AlanFoster** [comment]
@fernandes Check out the guard clause within the same function, you can supply your own React element for the header back button:

```
   _renderLeftComponent = (props: SceneProps) => {
     const options = this.props.getScreenDetails(props.scene).options;
     if (typeof options.headerLeft !== 'undefined') {
       return options.headerLeft;
     }
```

therefore if you want to handle your own logic in your own call back, this will work for you :+1:

**Comment by fernandes** [comment]
ohh @AlanFoster thank you for you awesome and fast reply!

I tried to overwrite the render left component, but I couldn't figure out how to do this. I just need to call:

`this.props.navigation.goBack()` instead of `this.props.navigation.goBack(null)` to work with my second stack, that's why I was trying to figure out how to overwrite the `_navigateBack` function (seemed easier at first sight).

I'm gonna try to overwrite the left element as you suggested. Thank you

**Comment by fernandes** [comment]
maybe improving my question: how can I use the default `HeaderBackButton` and just overwrite the `onPress?` any suggestion?

**Comment by AlanFoster** [comment]
@fernandes Sure:

```js
import React from 'react';
import { View, Text, Button } from 'react-native';
import { HeaderBackButton, StackNavigator } from 'react-navigation'

const ScreenOne = ({ navigation }) => (
  <View>
    <Text>Screen One</Text>
    <Button
      title="Open screen two!"
      onPress={() => navigation.navigate('ScreenTwo')}
    />
  </View>
);
const ScreenTwo = () => <Text>Screen Two</Text>;

const Application = StackNavigator({
  ScreenOne: {
    screen: ScreenOne,
    navigationOptions: {
      title: 'Screen One'
    }
  },
  ScreenTwo: {
    screen: ScreenTwo,
    navigationOptions({ navigation }) {
      return {
        title: 'Screen Two',
        headerLeft: (
          <HeaderBackButton
            title="Custom"
            onPress={() => navigation.goBack()}
          />
        )
      }
    }
  }
});

export default Application;
```

The relevant docs are:
- [Adding a right button](https://reactnavigation.org/docs/intro/headers#Adding-a-Right-Button) - But providing headerLeft isntead
- [navigation prop](https://reactnavigation.org/docs/navigators/navigation-prop)

Importantly, react-navigation exposes the HeaderBackButton that you want, and it can be imported via:

```js
import { HeaderBackButton, StackNavigator } from 'react-navigation'
```

You can see everything react-navigation exports via the docs, as well as the [source code](https://github.com/react-community/react-navigation/blob/ae9fb10e60d2735cd2debaae34483510aa9ab3e9/src/react-navigation.js#L7)

**Comment by fernandes** [comment]
hi @AlanFoster I do really appreciate your help.. everything is working perfectly, as I wired with redux for state management, on my back button I need to check the state to create the `Custom` back title, just replicating (borrowing) some `Header` logic, but it's a small cost for the benefits of nesting stack > drawer > stack and everything working fine.

thank you!

ps: Closing as I couldn't overwrite the `_navigateBack` directly, but @AlanFoster pointed another solution for my problem.
